tools/ports/sdl2_image.py

```diff
@@ -64,13 +80,20 @@ def clear(ports, settings, shared):
 
 def process_dependencies(settings):
   settings.USE_SDL = 2
-  if 'png' in settings.SDL2_IMAGE_FORMATS:
+  formats = get_formats(settings)
+  if 'png' in formats:
     deps.append('libpng')
     settings.USE_LIBPNG = 1
-  if 'jpg' in settings.SDL2_IMAGE_FORMATS:
+  if 'jpg' in formats:
     deps.append('libjpeg')
     settings.USE_LIBJPEG = 1
 
 
+def handle_options(options):
+  # options has been parsed from a query string
+  for fmts in options['formats']:
+    opts['formats'].update({format.lower() for format in re.findall(r'\b\w+\b', fmts)})
```

Comment:
Can we not just use `fmts.split(',')` here?

Reply:
Yes it would simplify this code, but then it would be a lot less flexible (for example, spaces or quotes would not be allowed), so it is a tradeoff. I will simplify it then.